src/app/core/session/auth/keycloak/auth-interceptor.service.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable } from "@angular/core";
+import {
+  HttpRequest,
+  HttpHandler,
+  HttpEvent,
+  HttpInterceptor,
+} from "@angular/common/http";
+import { Observable } from "rxjs";
+import { AuthService } from "../auth.service";
+
+@Injectable()
+export class AuthInterceptor implements HttpInterceptor {
```

Comment:
JSDocs? Also maybe briefly mention where/why the addAuthHeader function comes in as an alternative to be directly called?